lisa/node.py

```diff
@@ -114,6 +116,45 @@ def support_sudo(self) -> bool:
 
         return self._support_sudo
 
+    def check_sudo_password_required(self) -> None:
+        self.initialize()
+
+        # check if password is required when running command with sudo
+        if self.is_remote and self.is_posix:
+            process = self._execute("echo OK", shell=True, sudo=True, no_info_log=True)
+            result = process.wait_result(5)
+            if result.exit_code != 0 and "[sudo] password for" in result.stderr:
```

Comment:
stderr doesn't support in remote shell. So let's always use stdout. We need to copy stderr to stdout too.

Reply:
When it needs input of password, the process will be timed out, then the stdout of result comes from self._log_buffer. In this result, its stdout is "cmd: ['sudo', 'sh', '-c', 'echo OK'], cwd: None, shell: True...". So I added code below code in wait_result:

`if is_timeout:
                process_result = spur.results.result(
                    return_code=1,
                    allow_error=True,
                    output=self._log_buffer.getvalue(),
                    stderr_output="",
                )
                if not process_result.stderr_output:
                    # When the process is timedout, self._stdout_writer._buffer might
                    # have some information can tell the timeout reason
                    process_result.stderr_output = self._stdout_writer._buffer`